handlers/smshandler/jsonstorage.go

```diff
@@ -0,0 +1,291 @@
+package smshandler
+
+import (
+	"encoding/json"
+	"path/filepath"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/nyaruka/phonenumbers"
+	"github.com/vocdoni/blind-csp/types"
+	"go.vocdoni.io/dvote/db"
+	"go.vocdoni.io/dvote/db/metadb"
+	"go.vocdoni.io/dvote/log"
+)
+
+const authTokenIndexPrefix = "at_"
+
+type JSONstorage struct {
+	kv             db.Database
+	keysLock       sync.RWMutex
+	maxSmsAttempts int
+}
+
+func (js *JSONstorage) Init(dataDir string, maxAttempts int) error {
+	var err error
+	js.kv, err = metadb.New(db.TypePebble, filepath.Clean(dataDir))
+	if err != nil {
+		return err
+	}
+	js.maxSmsAttempts = maxAttempts
+	return nil
+}
+
+func (js *JSONstorage) AddUser(userID types.HexBytes, processIDs []types.HexBytes,
+	phone, extra string) error {
+	phoneNum, err := phonenumbers.Parse(phone, "ES")
+	if err != nil {
+		return err
+	}
+	js.keysLock.Lock()
+	defer js.keysLock.Unlock()
+	tx := js.kv.WriteTx()
+	defer tx.Discard()
+	maxAttempts := js.maxSmsAttempts * len(processIDs)
+	if maxAttempts == 0 {
+		maxAttempts = js.maxSmsAttempts
+	}
+	user := UserData{
+		Elections: []UserElection(HexBytesToElection(processIDs, js.maxSmsAttempts)),
+		ExtraData: extra,
+		Phone:     phoneNum,
+	}
+	userData, err := json.Marshal(user)
+	if err != nil {
+		return err
+	}
+	if err := tx.Set(userID, userData); err != nil {
+		return err
+	}
+	return tx.Commit()
+}
+
+func (js *JSONstorage) GetElections(userID types.HexBytes) ([]UserElection, error) {
```

Comment:
```suggestion
func (js *JSONstorage) Elections(userID types.HexBytes) ([]UserElection, error) {
```

https://go.dev/doc/effective_go#Getters